revisar-server-utils/middlewares.ts

```diff
@@ -30,3 +41,32 @@ export const schemaValidationMiddleware =
 
     return next();
   };
+
+// Add user property to express request interface
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace Express {
+    interface Request {
+      user?: IUserDetails;
+    }
+  }
+}
+
+export const authMiddleware = (req: AuthRequest, res: Response, next: NextFunction) => {
+  const token = extractBearerToken(req);
+  if (!token) return res.status(httpStatus.UNAUTHORIZED).send('No token provided');
+
+  try {
+    const user = <IUserDetails>jwt.verify(token, config.accessTokenSecret);
```

Comment:
@liadyacobi We can change to and remove the config.ts from revisar-server-utils
```suggestion
    const user = <IUserDetails>jwt.verify(token, config.accessTokenSecret);
```
```
    const user = <IUserDetails>jwt.verify(token, process.env.accessTokenSecret);
```

Reply:
I still don't think it's a good practice to require env variables in a package.
What we can do is create a function that receives a JWT SECRET and returns a middleware that uses this JWT SECRET to perform authentication.
This way we keep the logic in the package but the env variables in the services.
Something like:
```typescript
export const createAuthMiddleware = (jwtAccessToken: string) =>
   // ... Auth middleware code 
  (req: AuthRequest, res: Response, next: NextFunction) => {
    const token = extractBearerToken(req);
  };
```